tests/unit/user_interaction_test.py

```diff
@@ -0,0 +1,85 @@
+from unittest import TestCase
+import user_interaction
+from unittest.mock import patch
+import random
+import string
+
+
+class UserInteractionTest(TestCase):
+    def get_random_text(self):
+        return ''.join(random.choices(string.ascii_uppercase + string.digits, k=random.randint(1, 20)))
+
+    def print_test(self, func, text, test_text=None):
+        with patch("builtins.print") as mocked_print:
+            if test_text != None:
+                func(test_text)
+            else:
+                func()
+            mocked_print.assert_called_with(text)
+
+    def test_show_global_instant(self):
+        test_text = self.get_random_text()
+        self.print_test(user_interaction.show_global_instant,
+                        f"GLOBAL: {test_text}", test_text)
+
+    def test_show_active_instant(self):
+        test_text = self.get_random_text()
+        self.print_test(user_interaction.show_active_instant,
+                        f"ACTIVE: {test_text}", test_text)
+
+    def test_show_passive_instant(self):
+        test_text = self.get_random_text()
+        self.print_test(user_interaction.show_passive_instant,
+                        f"PASSIVE: {test_text}", test_text)
+
+    def test_show_all(self):
+        test_text_list = [self.get_random_text() for i in range(8)]
+
+        user_interaction.save_global(test_text_list[0])
+        user_interaction.save_active(test_text_list[1])
+        user_interaction.save_passive(test_text_list[2])
+        user_interaction.save_passive(test_text_list[3])
+        user_interaction.save_passive(test_text_list[4])
+        user_interaction.save_global(test_text_list[5])
+        user_interaction.save_active(test_text_list[6])
+        user_interaction.save_global(test_text_list[7])
+
+        asserted_text = f"{user_interaction.MessageScope.PASSIVE.name}: {str(test_text_list[2:5])}"
+
+        self.print_test(user_interaction.show_all, asserted_text)
+
+    def test_read_index(self):
+        test_int = random.randint(0, 1000)
+        test_to_string = str(test_int)
+
```

Comment:
To get the advantage of random value generation - it's better to generate bunch of different values (for example - in cycle). Otherwise we can get situation when tests are passing (because of luckily generated input) but user will receive an error during actual work.

Reply:
I divide my tests to valid and invalid data and iterate each of them 100 times.